### DISCUSSION TITLE
Why Subject is generic type?

### DISCUSSION THREAD

**Discussion by catcombo** [gleam-lang/gleam#3881] [Questions & support] [answered]
I'm not completely understand why `Subject` type is generic despite that generic argument is not used anywhere? Can someone explain it to me? I didn't find any explanation in the docs.

https://github.com/gleam-lang/erlang/blob/main/src/gleam/erlang/process.gleam#L64

What is the `Subject`'s generic part of the type when I call `new_subject()`?

For example, [in this example](https://github.com/GauBen/real-time-gleam-chat/blob/main/src/real_time_chat.gleam#L21) `Subject` has a `String` generic type, but it doesn't make any sense. I totally confused how it works.

**Comment by GearsDatapacks** [comment]
It's called a phantom type. Even though a value of that type isn't stored in the value itself, it determines what kind of values should be used in combination with it. I would recommend reading [Hayleigh's blog post](https://blog.hayleigh.dev/phantom-types-in-gleam) on phantom types.

**Reply by lpil** [reply, marked as answer]
That's it!

If it didn't have a type parameter there would be no way for the subject to specify what type of message it accepts.

**Reply by catcombo** [reply]
Thank you very much for explanation!
